File: testcode/VCOSweep.py
import smbus

# I2C channel 1 is connected to the GPIO pins
channel = 1

#  MCP4725 defaults to address 0x60
address = 0x4C

# Register addresses (with "normal mode" power-down bits)
reg_write_dac = 0x16 #0b00010110 - This writes to the first output port on the DAC

# Initialize I2C (SMBus)
bus = smbus.SMBus(channel)

# Create our 12-bit number representing relative voltage
# Voltage   Frequency
# 0x55F     5.65
# 0x7CF     5.8
# 0xABF     5.95
voltage = 0x7CF

while True:

    for i in range(0x55F,0xABF):
        msg = [((i & 0xff0) >> 4& 0xf), ((i & 0xff0) >> 4& 0xf) << 4]

        # Write out I2C command: address, reg_write_dac, msg[0], msg[1]
        bus.write_i2c_block_data(address, reg_write_dac, msg)

File: testcode/vco-sweep.c
#include <string.h>
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <linux/i2c-dev.h>
#include <sys/ioctl.h>
#include <sys/types.h>
#include <fcntl.h>

void main (void) {
    int file;
    char filename[40];
    int addr = 0x4C; // The I2C address of the DAC

    sprintf(filename,"/dev/i2c-1");
    if ((file = open(filename,O_RDWR)) < 0) {
        printf("Failed to open the bus.");
        exit(1);
    }

    if (ioctl(file,I2C_SLAVE,addr) < 0) {
        printf("Failed to acquire bus access and/or talk to slave.\n");
        exit(1);
    }

    char buf[3] = {0};
    buf[0] = 0x16; // Register we are writing to
    while (1){
        for(int i = 0x55F; i < 0xABF; i += 8){
            buf[1] = (i & 0xff0) >> 4;
            buf[2] = (buf[1] & 0xf) << 4;
            write(file,buf,3);
        }
    }
}
